dynamodb/dynamo_lock_table.go

```diff
@@ -204,3 +215,53 @@ type TableDoesNotExist struct {
 func (err TableDoesNotExist) Error() string {
 	return fmt.Sprintf("Table %s does not exist in DynamoDB! Original error from AWS: %v", err.TableName, err.Underlying)
 }
+
+// Encrypt the TFState Lock table - If Necessary
+func UpdateLockTableSetSSEncryptionOnIfNecessary(tableName string, client *dynamodb.DynamoDB, terragruntOptions *options.TerragruntOptions) error {
+	tableSSEncrypted, err := LockTableCheckSSEncryptionIsOn(tableName, client)
+	if err != nil {
+		return err
+	}
+
+	if !tableSSEncrypted {
+		tableCreateDeleteSemaphore.Acquire()
+		defer tableCreateDeleteSemaphore.Release()
+
+		terragruntOptions.Logger.Printf("Setting server-side encryption on table %s in DynamoDB", tableName)
+
+		input := &dynamodb.UpdateTableInput{
+			SSESpecification: &dynamodb.SSESpecification{
+				Enabled: aws.Bool(true),
+				SSEType: aws.String("KMS"),
+			},
+			TableName: aws.String(tableName),
+		}
+
+		result, err := client.UpdateTable(input)
+		if err != nil {
+			if aerr, ok := err.(awserr.Error); ok {
+				switch aerr.Code() {
+				case dynamodb.ErrCodeResourceInUseException:
+					fmt.Println(dynamodb.ErrCodeResourceInUseException, aerr.Error())
+				case dynamodb.ErrCodeResourceNotFoundException:
+					fmt.Println(dynamodb.ErrCodeResourceNotFoundException, aerr.Error())
+				case dynamodb.ErrCodeLimitExceededException:
+					fmt.Println(dynamodb.ErrCodeLimitExceededException, aerr.Error())
+				case dynamodb.ErrCodeInternalServerError:
+					fmt.Println(dynamodb.ErrCodeInternalServerError, aerr.Error())
+				default:
+					fmt.Println(aerr.Error())
```

Comment:
Why all these `fmt.Println`? If these are just for debugging, please remove, and just return the `err`.

Reply:
@brikis98 OK - done. Will commit in a couple of minutes.

Reply:
@brikis98 Done.